File: frontend/allfolio_app/types/exclusion-list.ts
// types/exclusion-list.ts
export interface ExclusionItem {
  id: string
  symbol: string
  memo: string | null
  addedAt: string
}

export interface ExclusionList {
  id: string
  name: string
  category: string
  description: string | null
  active: boolean
  itemCount: number
  items: ExclusionItem[]
  updatedAt: string
}

export interface PresetSymbol { symbol: string; reason: string }
export interface Preset { name: string; symbols: PresetSymbol[] }

export interface CreateList { name: string; category: string; description?: string | null }
export interface UpdateList { name: string; category: string; description?: string | null; active: boolean }

File: allfolio-backend/unified-asset/src/main/kotlin/com/allfolio/unifiedasset/domain/exclusion/ExclusionList.kt
package com.allfolio.unifiedasset.domain.exclusion

import java.time.LocalDateTime
import java.util.UUID

data class ExclusionList(
    val id: UUID,
    val userId: UUID,
    val name: String,
    val category: String,
    val description: String?,
    val active: Boolean,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val items: List<ExclusionItem> = emptyList(),
)
